src/colorizer/utils/math_utils.ts

```diff
@@ -93,6 +93,64 @@ export function numberToSciNotation(input: number, significantFigures: number):
   return `${prefix}${coefficient.toFixed(significantFigures - 1)}×10${numberToUnicodeSuperscript(exponent)}`;
 }
 
+/**
+ * Calculates the size of a frame in pixels that is scaled to fit within a canvas with known
+ * onscreen pixel dimensions.
+ * @param canvasSizePx Size of the canvas, in pixels.
+ * @param frameResolution Resolution of the frame, in pixels or units.
+ * @param frameZoom The zoom level of the frame. A zoom of 1x means the frame is scaled to fit in the canvas
+ * while maintaining its aspect ratio. A zoom of 2x means the frame is twice as large as it would be at 1x zoom.
+ * @returns A tuple of `[width, height]` in pixels.
+ */
+export function getFrameSizeInScreenPx(
+  canvasSizePx: [number, number] | number[],
+  frameResolution: [number, number] | number[],
+  frameZoom: number
+): [number, number] {
```

Comment:
Oops, I think I left the `| number[]` in while I was fighting with Typescript since it wasn't recognizing number tuples by default.

I'll play with this and see how much work it is to replace the tuples with Vector2.

Reply:
Done!